modules/oe_content_persistent/src/Controller/PersistentUrlController.php

```diff
@@ -65,16 +72,17 @@ public static function create(ContainerInterface $container) {
    *   A redirect response to actual alias or system path.
    */
   public function index(string $uuid): RedirectResponse {
-    if ($entity = $this->contentUuidResolver->getEntityByUuid($uuid)) {
-      // Unfortunately we cannot use
-      // an instance of CacheableSecuredRedirectResponse because we get
-      // an exception inside
-      // \Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber.
-      // More information you could find in this article:
-      // https://www.lullabot.com/articles/early-rendering-a-lesson-in-debugging-drupal-8
+    $langcode = $this->languageManager->getCurrentLanguage()->getId();
+    if ($entity = $this->contentUuidResolver->getEntityByUuid($uuid, $langcode)) {
       if ($entity instanceof ContentEntityInterface) {
         $url = $this->contentUrlResolver->resolveUrl($entity);
-        return new RedirectResponse($url->toString(), 302, ['PURL' => '1']);
+        $generated_url = $url->toString(TRUE);
+        $cache = CacheableMetadata::createFromObject($generated_url);
+        $cache->addCacheableDependency($entity);
+        $cache->addCacheContexts(['url', 'languages']);
```

Comment:
Probably better to use a more specific cache context like `url.path`.

Reply:
It's ok, let's keep it as wide as possible.